### ISSUE TITLE
update container/helm spec ingestion and validation function

### ISSUE THREAD

**Issue by mitchdraft** [solo-io/build#9]
following on https://github.com/solo-io/build/pull/7 and discussion with @marcogschmidt and @rickducott 

container specs should be optional
helm specs should be optional

~if either is specified, both test and release values should be provided~
you can provide one or the other or both. During the test stage we look for a test configuration and ~nop~ *return NOT_SPECIFIED* if none provided; during release stage we look only at release.

change the validation function to take arguments from flags, not from an args list so that empty values may be passed

**Comment by rickducott** [comment]
_if either is specified, both test and release values should be provided_

I was actually thinking you could provide one or the other or both. During the test stage we look for a test configuration and nop if none provided; during release stage we look only at release.

**Comment by mitchdraft** [comment]
sounds good, updated, I think we want to return a known string in the case of unset values, let me know what you think

**Comment by rickducott** [comment]
In what context? in the parse-env command, that's fine to return NOT SPECIFIED instead of an empty string for a nil config. i don't think that's what you're referring to here though

**Comment by mitchdraft** [comment]
yep, that's what I mean. The [computed values](https://github.com/solo-io/build/blob/master/api/v1/build.proto#L22) should be set to a known string, like "NOT_SPECIFIED" if the user has not provided the config values required for the computation.

I think this will increase transparency regarding the build tool's perspective on the config/env
